sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala

```diff
@@ -301,6 +305,11 @@ class SparkSessionExtensionSuite extends SparkFunSuite with SQLHelper {
       assert(result(0).getLong(0) == 101L) // Check that broken columnar Add was used.
       assert(result(1).getLong(0) == 201L)
       assert(result(2).getLong(0) == 301L)
+
+      withTempPath { path =>
+        val e = intercept[Exception](df.write.parquet(path.getCanonicalPath))
```

Comment:
without the fix, it fails with `class org.apache.spark.sql.execution.WholeStageCodegenExec has write support mismatch`

Reply:
Can we catch more specific exception though?

Reply:
This is the exact match: https://github.com/apache/spark/pull/39922/files#diff-d8405c694c782a5adf4bdefcd723f8284e2a3250ac20ff4609163b4caeb29870R815

Reply:
Got it. Since it's a test suite and this test case matches the exception message exactly, looks fine.